### ISSUE TITLE
[Support] Could not fetch X, please create a GitHub OAuth token

### ISSUE THREAD

**Issue by kinglozzer** [composer/composer#6614]
Hi,

I’m trying to run a `composer update`, and am hitting the following error:

```
Could not fetch https://api.github.com/repos/kinglozzer/silverstripe-importexport, please create a GitHub OAuth token to go over the API rate limit
Head to https://github.com/settings/tokens/new?scopes=repo&description=Composer+on+loz.local+2017-08-16+1558
```

I’ve tried:

- Regenerating the API token and pasting it when prompted;
- Adding the new token via `composer config --global --auth github-oauth.github.com <token>`;
- Adding the new token by editing `~/.composer/auth.json` manually;
- Adding the new token to my project’s `composer.json` directly.

I’ve had no luck with any of the above. The repo definitely exists, and the URL it’s failing to fetch seems to be fine too.

Any suggestions?

```json
{
    "require": {
        "burnbright/silverstripe-importexport": "dev-misc-additions#4f7318d",
        "... other deps"
    },
     "repositories": [
        {
            "type": "vcs",
            "url": "git@github.com:kinglozzer/silverstripe-importexport.git"
        }
    ],
    "config": {
        "github-oauth": {
            "github.com": "<token>"
        }
    }
}
```

**Comment by kinglozzer** [comment]
Would you believe it? As soon as I post the issue, the next `composer update` works perfectly 😅.

I’m guessing either Composer had cached something, or Github was still rejecting requests despite the new token (perhaps based on user-agent? As visiting the URL in-browser from the same IP address worked fine).

¯\\_(ツ)_/¯

**Comment by curry684** [comment]
The full error you're copying is:

> Could not fetch https://api.github.com/repos/kinglozzer/silverstripe-importexport, please create a GitHub OAuth token to go over the API rate limit

As clearly indicated there you had exceeded the [GitHub API Rate Limit](https://developer.github.com/v3/rate_limit/). By creating an OAuth token as suggested you can increase the limits tenfold if you run into this again.

It's [trivial to check your current rate limit](https://developer.github.com/v3/#rate-limiting) and how fast you're burning it.

**Comment by kinglozzer** [comment]
Indeed, but as per my description I’d created an OAuth token and added it to my config, but was still hitting the rate limit.

Thanks for the second link, that would’ve been useful for confirming the issue!

**Comment by lorenzomormina** [comment]
Today I got this error, but I would like to understand a few things. I created today a private repository and added a collaborator. I can do "composer install" just fine, but he gets this message. Also I must have used GitHub more than him lately.

Help me understand why I don't get the error while he does. I would be curious to see if he gets the error again if he deleted the .git folder (it might be a stupid idea, but I thought that he was like penalized for being only a collaborator or something).

This message awaits for the user to paste the token in the console, but he noticed that he had to paste it for every package he was downloading during the same "composer install".

Is there any solution/workaround?
